src/class_sqlThread.py

```diff
@@ -629,11 +401,158 @@ def run(self):  # pylint: disable=too-many-locals, too-many-branches, too-many-s
                 helper_sql.sqlReturnQueue.put((self.cur.fetchall(), rowcount))
                 # helper_sql.sqlSubmitQueue.task_done()
 
-    def create_function(self):
-        # create_function
+    def earlier_setting_version(self, settingsversion):
+        """
+            Upgrade schema with respect setting version
+        """
+
+        # People running earlier versions of PyBitmessage do not have the
+        # usedpersonally field in their pubkeys table. Let's add it.
+        if settingsversion == 2:
+            item = '''ALTER TABLE pubkeys ADD usedpersonally text DEFAULT 'no' '''
+            parameters = ()
+            self.cur.execute(item, parameters)
+            self.conn.commit()
+
+            settingsversion = 3
+
+        # People running earlier versions of PyBitmessage do not have the
+        # encodingtype field in their inbox and sent tables or the read field
+        # in the inbox table. Let's add them.
+        if settingsversion == 3:
+            item = '''ALTER TABLE inbox ADD encodingtype int DEFAULT '2' '''
+            parameters = ()
+            self.cur.execute(item, parameters)
+
+            item = '''ALTER TABLE inbox ADD read bool DEFAULT '1' '''
+            parameters = ()
+            self.cur.execute(item, parameters)
+
+            item = '''ALTER TABLE sent ADD encodingtype int DEFAULT '2' '''
+            parameters = ()
+            self.cur.execute(item, parameters)
+            self.conn.commit()
+
+            return 4
+        return settingsversion
+
+
+class sqlThread(threading.Thread, UpgradeDB):
+    """A thread for all SQL operations"""
+
+    def __init__(self):
+        super(sqlThread, self).__init__()
+        threading.Thread.__init__(self, name="SQL")
+        self.__flag = threading.Event()
+
+        # Adding for pause/resume thread
+        self.__flag = threading.Event()
+        self.__flag.set()  # Set to True
+        self.__running = threading.Event()  # Used to stop the thread identification
+        self.__running.set()  # Set running to True
+
+    def run(self):  # pylint: disable=too-many-locals, too-many-branches, too-many-statements,
+        # Redefinition-of-parameters-type-from-tuple-to-str, R0204, line-too-long, E501
+        """Process SQL queries from `.helper_sql.sqlSubmitQueue`"""
+        helper_sql.sql_available = True
+        self.conn, self.cur = UpgradeDB().conn, UpgradeDB().cur
+
+        self.cur.execute('PRAGMA secure_delete = true')
+
+        # call create_function for encode address
+        self.create_function(self.conn)
+
+        self.initiate_database(self.cur, self.conn)
```

Comment:
Something like this to skip init when running tests:
```
if not self.instanceof(MockDB):
   self.initiate_database()
```

Reply:
managed in earlier logic used in memory for test mocking